synapse/rest/client/v2_alpha/sync.py

```diff
@@ -118,8 +120,14 @@ def on_GET(self, request):
             except:
                 filter = FilterCollection({})
 
+        if is_guest and filter.list_rooms() is None:
```

Comment:
It looks like `filter.list_rooms()` will return `None` iff there is no `rooms` key but there is no 0-length array check. We should probably guard against `{ "rooms": [] }` as well.